### PR TITLE
Tidy Up Ram and avoid pointers in Aggregates

### PR DESCRIPTION
**Pull request by b-scholz**
This pull request is related to issue #1000. 

Null-pointers are used for various semantic aspects and have been replaced by proper values in RAM such as true/false and undefined values. 

The RamNode code was tidied up and default behavior added so that we can reduce code size.

### CODE REVIEW

File: src/RamOperation.h [hunk 1]
```diff
@@ -36,22 +36,14 @@ namespace souffle {
  */
 class RamOperation : public RamNode {
 public:
-    RamOperation() = default;
-
+    /** pretty print with intentation */
```
Comment (mmcgr) [inline src/RamOperation.h]:
```suggestion
    /** pretty print with indentation */
```

**Comment by b-scholz** [comment]
Jenkins test please.

**Comment by b-scholz** [comment]
Jenko, my good mate, test this please

**Comment by mmcgr** [comment]
Seems there were network issues somewhere between jenkins and github - also jenkins doesn't handle some git network issues well. So, Jenkins, test this please.

**Pull request merged**